core/lib/resources/presence/presence_manager.js

```diff
@@ -39,13 +40,30 @@ PresenceManager.prototype.setup = function() {
   });
 
   // save so you removeListener on destroy
-  this.sentryListener = function(sentry) {
+  this.sentryListener = function (sentry) {
+    if (manager.destroying) return;
+
     var clientIds = store.clientsForSentry(sentry);
 
     var chain = function (clientIds) {
+      var immediateObject;
       var clientId = clientIds.pop();
 
-      if (!clientId) return;
+      // Pick up any new users/clients that have subscribed to this resource
+      if (!clientId && !manager.destroying) {
+        clientIds = store.clientsForSentry(sentry);
```

Comment:
This line is wrong. We should not reassign the clientIds array here. It should be set only in the beginning. As it passes through the chain, it keeps popping one client at a time.